src/Symfony/Bundle/FrameworkBundle/Test/BrowserKitAssertionsTrait.php

```diff
@@ -124,6 +126,19 @@ public static function assertRouteSame($expectedRoute, array $parameters = [], s
         self::assertThat(self::getRequest(), $constraint, $message);
     }
 
+    public static function assertThatForResponse(Response $response, Constraint $constraint, string $message = ''): void
+    {
+        try {
+            self::assertThat($response, $constraint, $message);
+        } catch (ExpectationFailedException $exception) {
+            if ($response->headers->has('X-Debug-Exception')) {
+                throw new \Exception($response->headers->get('X-Debug-Exception'));
```

Comment:
I would expect an `ExpectationFailedException` to be thrown.
we should also preserve the previous exception.
and we could merge the previous and the `X-Debug-Exception` message.

Reply:
Ok, done. I create a new `ExpectationFailedException` (not possible to just re set the message of an existing one) with merged messages.